diff --git a/node_modules/react-native-reanimated/android/src/main/cpp/reanimated/CMakeLists.txt b/node_modules/react-native-reanimated/android/src/main/cpp/reanimated/CMakeLists.txt
index e8267fe..a4066ec 100644
--- a/node_modules/react-native-reanimated/android/src/main/cpp/reanimated/CMakeLists.txt
+++ b/node_modules/react-native-reanimated/android/src/main/cpp/reanimated/CMakeLists.txt
@@ -37,6 +37,8 @@ set_target_properties(
     PROPERTIES
     LINKER_LANGUAGE
     CXX
+    LINK_FLAGS
+    "-Wl,-z,max-page-size=16384"
 )
 
 target_link_libraries(
diff --git a/node_modules/react-native-reanimated/android/src/main/cpp/worklets/CMakeLists.txt b/node_modules/react-native-reanimated/android/src/main/cpp/worklets/CMakeLists.txt
index 61508ef..dedcb0f 100644
--- a/node_modules/react-native-reanimated/android/src/main/cpp/worklets/CMakeLists.txt
+++ b/node_modules/react-native-reanimated/android/src/main/cpp/worklets/CMakeLists.txt
@@ -57,6 +57,8 @@ set_target_properties(
     PROPERTIES
     LINKER_LANGUAGE
     CXX
+    LINK_FLAGS
+    "-Wl,-z,max-page-size=16384"
 )
 
 target_link_libraries(
